hr_employee_exemption/tests/test_hr_employee_exemption.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding:utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Savoir-faire Linux. All Rights Reserved.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.tests import common
+
+
+class test_hr_employee_exemption(common.TransactionCase):
+    def setUp(self):
+        super(test_hr_employee_exemption, self).setUp()
+        self.employee_model = self.registry('hr.employee')
+        self.exemption_model = self.registry('hr.income.tax.exemption')
+        self.rule_model = self.registry('hr.salary.rule')
+        self.rule_category_model = self.registry("hr.salary.rule.category")
+        self.contract_model = self.registry('hr.contract')
+        self.structure_model = self.registry("hr.payroll.structure")
+        self.payslip_model = self.registry('hr.payslip')
+        self.user_model = self.registry("res.users")
```

Comment:
replace self.registry(...) with self.env[...] and remove cr, uid, context all over the file.